### PR TITLE
Remove plugin/pkg/auth/* from hack/.golint_failures

### PR DESCRIPTION
**Pull request by emwalker**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes golint errors in several files under plugin/pkg/auth/.

**Which issue(s) this PR fixes**:
See #68026 for more information.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/auth/authorizer/node/graph.go [hunk 1]
```diff
@@ -132,21 +133,21 @@ var vertexTypes = map[vertexType]string{
 }
 
 // must be called under a write lock
```
Comment (mourya007) [inline plugin/pkg/auth/authorizer/node/graph.go]: getOrCreateVertexLocked() should also  get specified under the comments before function name.

File: plugin/pkg/auth/authorizer/node/graph.go [hunk 2]
```diff
@@ -132,21 +133,21 @@ var vertexTypes = map[vertexType]string{
...
+	if vertex, exists := g.getVertexRlocked(vertexType, namespace, name); exists {
 		return vertex
 	}
-	return g.createVertex_locked(vertexType, namespace, name)
+	return g.createVertexLocked(vertexType, namespace, name)
 }
 
 // must be called under a read lock
-func (g *Graph) getVertex_rlocked(vertexType vertexType, namespace, name string) (*namedVertex, bool) {
+func (g *Graph) getVertexRlocked(vertexType vertexType, namespace, name string) (*namedVertex, bool) {
```
Comment (mourya007) [inline plugin/pkg/auth/authorizer/node/graph.go]: ditto

File: plugin/pkg/auth/authorizer/node/graph.go [hunk 3]
```diff
@@ -132,21 +133,21 @@ var vertexTypes = map[vertexType]string{
...
 	vertex, exists := g.vertices[vertexType][namespace][name]
 	return vertex, exists
 }
 
 // must be called under a write lock
-func (g *Graph) createVertex_locked(vertexType vertexType, namespace, name string) *namedVertex {
+func (g *Graph) createVertexLocked(vertexType vertexType, namespace, name string) *namedVertex {
```
Comment (mourya007) [inline plugin/pkg/auth/authorizer/node/graph.go]: ditto

File: plugin/pkg/auth/authorizer/node/graph.go [hunk 4]
```diff
@@ -167,8 +168,8 @@ func (g *Graph) createVertex_locked(vertexType vertexType, namespace, name strin
 }
 
 // must be called under write lock
-func (g *Graph) deleteVertex_locked(vertexType vertexType, namespace, name string) {
-	vertex, exists := g.getVertex_rlocked(vertexType, namespace, name)
+func (g *Graph) deleteVertexLocked(vertexType vertexType, namespace, name string) {
```
Comment (mourya007) [inline plugin/pkg/auth/authorizer/node/graph.go]: ditto

File: plugin/pkg/auth/authorizer/rbac/rbac_test.go [hunk 5]
```diff
@@ -246,7 +246,7 @@ func TestAuthorizer(t *testing.T) {
 	}
 	for i, tt := range tests {
 		ruleResolver, _ := rbacregistryvalidation.NewTestRuleResolver(tt.roles, tt.roleBindings, tt.clusterRoles, tt.clusterRoleBindings)
-		a := RBACAuthorizer{ruleResolver}
+		a := Authorizer{ruleResolver}
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/rbac/rbac_test.go]: -1 on type change here and for NodeAuthorizer

File: plugin/pkg/auth/authorizer/node/graph_test.go [hunk 6]
```diff
@@ -124,19 +124,19 @@ func TestDeleteEdges_locked(t *testing.T) {
...
-				g.getOrCreateVertex_locked(nodeVertexType, "", "node1")
+				g.getOrCreateVertexLocked(nodeVertexType, "", "node1")
 				return g
 			}(),
 		},
 	}
 	for _, c := range cases {
 		t.Run(c.desc, func(t *testing.T) {
-			c.start.deleteEdges_locked(c.fromType, c.toType, c.toNamespace, c.toName)
+			c.start.deleteEdgesLocked(c.fromType, c.toType, c.toNamespace, c.toName)
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/graph_test.go]: -1 on merging the locked indicator into the name... in my opinion, the _ separator makes the intent clearer

File: plugin/pkg/auth/authorizer/node/authorizer_test.go [hunk 7]
```diff
@@ -101,7 +101,7 @@ func TestAuthorizer(t *testing.T) {
 	populate(g, nodes, pods, pvs, attachments)
 
 	identifier := nodeidentifier.NewDefaultNodeIdentifier()
-	authz := NewAuthorizer(g, identifier, bootstrappolicy.NodeRules()).(*NodeAuthorizer)
+	authz := NewAuthorizer(g, identifier, bootstrappolicy.NodeRules()).(*Authorizer)
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/authorizer_test.go]: please revert to NodeAuthorizer

File: plugin/pkg/auth/authorizer/node/BUILD [hunk 8]
```diff
@@ -35,7 +35,7 @@ go_library(
         "graph.go",
         "graph_populator.go",
         "intset.go",
-        "node_authorizer.go",
+        "authorizer.go",
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/BUILD]: please revert these filenames... specific filenames allows using --vmodule to increase/decrease log levels just for this component

**Comment by liggitt** [comment]
the public godoc improvements are good, thanks for making those

However, the other changes made to pass lint checks actually reduce readability and aspects of control over logging. I don't think making those changes is worth it just to make lint pass.

**Comment by emwalker** [comment]
> However, the other changes made to pass lint checks actually reduce readability and aspects of control over logging. I don't think making those changes is worth it just to make lint pass.

No worries — I figured some of them were definitely questionable.  I wish the linter was configurable.  I'll abandon this PR.

**Pull request closed without merging**